gripper/firmware/interfaces.cpp

```diff
@@ -0,0 +1,166 @@
+#pragma GCC diagnostic ignored "-Wvolatile"
+#include "motor_hardware.h"
+#pragma GCC diagnostic pop
+
+static auto iWatchdog = iwdg::IndependentWatchDog{};
+
+/**
+ * The SPI configuration.
+ */
+static spi::SPI_interface SPI_intf = {
+    .SPI_handle = &hspi2,
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+    .GPIO_handle = GPIOB,
+    .pin = GPIO_PIN_12,
+};
+
+/**
+ * The SPI interface.
+ */
+static spi::Spi spi_comms(SPI_intf);
+
+/**
+ * Motor pin configuration.
+ */
+struct motion_controller::HardwareConfig motor_pins {
+    .direction =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOB,
+            .pin = GPIO_PIN_10,
+            .active_setting = GPIO_PIN_RESET},
+    .step =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOB,
+            .pin = GPIO_PIN_1,
+            .active_setting = GPIO_PIN_SET},
+    .enable =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOA,
+            .pin = GPIO_PIN_9,
+            .active_setting = GPIO_PIN_SET},
+    .limit_switch =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOC,
+            .pin = GPIO_PIN_7,
+            .active_setting = GPIO_PIN_SET},
+    .led = {
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+        .port = GPIOC,
+        .pin = GPIO_PIN_6,
+        .active_setting = GPIO_PIN_RESET},
+};
+
+/**
+ * The motor hardware interface.
+ */
+static motor_hardware::MotorHardware motor_hardware_iface(motor_pins, &htim7);
+
+/**
+ * Motor driver configuration.
+ */
+static tmc2130::TMC2130DriverConfig MotorDriverConfigurations{
+    .registers =
+        {
+            .gconfig = {.en_pwm_mode = 1},
+            .ihold_irun = {.hold_current = 0x2,
+                           .run_current = 0x19,
+                           .hold_current_delay = 0x7},
+            .tpowerdown = {},
+            .tcoolthrs = {.threshold = 0},
+            .thigh = {.threshold = 0xFFFFF},
+            .chopconf = {.toff = 0x5,
+                         .hstrt = 0x5,
+                         .hend = 0x3,
+                         .tbl = 0x2,
+                         .mres = 0x4},
+            .coolconf = {.sgt = 0x6},
+        },
+    .current_config = {
+        .r_sense = 0.1,
+        .v_sf = 0.325,
+    }};
+
+/**
+ * The can bus.
+ */
+static auto canbus = hal_can_bus::HalCanBus(can_get_device_handle());
+
+/**
+ * The pending move queue
+ */
+static freertos_message_queue::FreeRTOSMessageQueue<motor_messages::Move>
+    motor_queue("Motor Queue");
+
+/**
+ * The motor struct.
+ */
+static motor_class::Motor motor{
+    spi_comms,
+    lms::LinearMotionSystemConfig<lms::BeltConfig>{
+        .mech_config =
+            lms::BeltConfig{.belt_pitch = 2, .pulley_tooth_count = 20},
```

Comment:
it's a lead screw isn't it?

Reply:
yes will change it once i get the vals from hardware